bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/AbstractTableViewer.java

```diff
@@ -666,9 +687,13 @@ private void internalRefreshAll(boolean updateLabels) {
 		// e.g. if (a, b) is replaced by (b, a), the disassociate of b to
 		// item 1 could undo
 		// the associate of b to item 0.
-
-		Object[] children = getSortedChildren(getRoot());
+		Object[] children = null;
 		Item[] items = doGetItems();
+		if (getItemsLimit() > 0 && items.length > getItemsLimit()) {
```

Comment:
This check duplicates one in getVisibleLimitBasedChildren. Additionally, getVisibleLimitBasedChildren can return null but that is not checked here